src/Symfony/Bundle/FrameworkBundle/Test/MessengerAssertionsTrait.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use Symfony\Component\Messenger\EventListener\MessagesSentToTransportsListener;
+use Symfony\Component\Messenger\Test\Constraint as MessengerConstraint;
+
+/*
+ * @author Marilena Ruffelaere <marilena.ruffelaere@gmail.com>
+ */
+
+trait MessengerAssertionsTrait
+{
+    /**
+     * @param int         $count   the expected number of messages
+     * @param string|null $busName The busName to consider. If null all the collected messages will be counted.
+     */
+    public static function assertMessagesByBusCount(int $count, string $busName = null, string $message = ''): void
```

Comment:
Naming : I'd go for a `assertCountMessages(int $count, string $busName = null, string $message = '')` or a `assertMessagesCount(int $count, string $busName = null, string $message = '')` or `assertPendingMessagesCount(int $count, string $busName = null, string $message = '')`

Reply:
Ok, `assertMessagesCount(int $count, string $busName = null, string $message = '') ` seems nice to me